Reskin card type chooser according to designs

burieberry opened the pull request:
[CS-5445](https://linear.app/cardstack/issue/CS-5445/re-skin-card-type-chooser)

- Can select card
- Buttons display on hover
- Added footer with cardURL input, cancel and go buttons
- Update tests to use the go button after card selection

<img width="976" alt="card-type-catalog" src="https://github.com/cardstack/boxel/assets/16160806/c017e979-b605-4132-a2e5-0fdbb11addf2">

burieberry committed:
Make card selection more consistent with designs

burieberry committed:
Merge remote-tracking branch 'origin/main' into cs-5445-reskin-card-chooser

burieberry committed:
Hover state buttons and styling

burieberry committed:
Display result count

burieberry committed:
Add cardURL input box in footer

burieberry committed:
Update tests

jurgenwerk commented on packages/host/app/components/card-catalog-modal.gts:
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
+                  this.currentRequest.search.instances.length
+                  '0'
+                }}
+                results</div>
```
There can also be only 1 result and we should show the singular form of "results" in that case

jurgenwerk replied:
Also I think "results" shouldn't be shown without a number in front of it, so either "x results" or nothing. If I understand correctly currently with this code there will be a moment where only "results" will be shown when there is no number yet?

jurgenwerk reviewed (approved):
Minor pluralization thing but otherwise looks good

jurgenwerk commented on packages/host/app/components/card-catalog-modal.gts:
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
+                  <li
+                    class={{cn
+                      'card-catalog-item'
+                      is-selected=(eq this.selectedCard.id card.id)
```
Maybe modifier css class would be more appropriate? `card-catalog-item--selected`

lukemelia commented on packages/host/app/components/card-catalog-modal.gts:
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
                     <button
-                      {{on 'click' (fn this.pick card)}}
+                      class='card-catalog-item__select'
+                      {{on 'click' (fn this.select card)}}
```
Can you unselect a card once it is selected?

lukemelia commented on packages/host/app/components/card-catalog-modal.gts:
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
+              <Button
+                @kind='secondary-light'
+                @size='tall'
+                @disabled={{eq this.selectedCard undefined}}
```
It seems like the Cancel button should still be available even if no card has been selected.

lukemelia reviewed (commented):
Perhaps doubleclick on a card should pick it, without having to press Go?

burieberry committed:
Merge remote-tracking branch 'origin/main' into cs-5445-reskin-card-chooser

burieberry committed:
Review updates

- Only display results count if there's 1 or more
- Adjust results display for singular result
- Can select or unselect cards
- Do not disable "cancel" button

burieberry committed:
Merge remote-tracking branch 'origin/main' into cs-5445-reskin-card-chooser

burieberry committed:
Update tests

burieberry committed:
Update test helpers

burieberry committed:
Merge remote-tracking branch 'origin/main' into cs-5445-reskin-card-chooser

burieberry committed:
Update test helper
